broker/src/test/java/io/camunda/zeebe/broker/system/partitions/impl/steps/ExporterDirectorPartitionTransitionStepTest.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.system.partitions.impl.steps;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.atomix.raft.RaftServer.Role;
+import io.camunda.zeebe.broker.exporter.repo.ExporterRepository;
+import io.camunda.zeebe.broker.exporter.stream.ExporterDirector;
+import io.camunda.zeebe.broker.system.partitions.TestPartitionTransitionContext;
+import io.camunda.zeebe.logstreams.log.LogStream;
+import io.camunda.zeebe.util.health.HealthMonitor;
+import io.camunda.zeebe.util.sched.ActorSchedulingService;
+import io.camunda.zeebe.util.sched.future.TestActorFuture;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class ExporterDirectorPartitionTransitionStepTest {
+
+  final TestPartitionTransitionContext transitionContext = new TestPartitionTransitionContext();
+
+  final ActorSchedulingService actorSchedulingService = mock(ActorSchedulingService.class);
+  final ExporterRepository exporterRepository = mock(ExporterRepository.class);
+  final ExporterDirector exporterDirectorFromPrevRole = mock(ExporterDirector.class);
+  private ExporterDirectorPartitionTransitionStep step;
+
+  @BeforeEach
+  void setup() {
+    transitionContext.setLogStream(mock(LogStream.class));
+    transitionContext.setComponentHealthMonitor(mock(HealthMonitor.class));
+
+    when(exporterRepository.getExporters()).thenReturn(Map.of());
+    transitionContext.setExporterRepository(exporterRepository);
+
+    when(actorSchedulingService.submitActor(any(), anyInt()))
+        .thenReturn(TestActorFuture.completedFuture(null));
+    transitionContext.setActorSchedulingService(actorSchedulingService);
+
+    when(exporterDirectorFromPrevRole.closeAsync())
+        .thenReturn(TestActorFuture.completedFuture(null));
+
+    step = new ExporterDirectorPartitionTransitionStep();
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideTransitionsThatShouldCloseExistingExporter")
+  void shouldCloseExistingStreamProcessor(final Role currentRole, final Role targetRole) {
+    // given
+    transitionContext.setCurrentRole(currentRole);
```

Comment:
:thought_balloon: These four lines are repreated a couple of times. Could be refactored into a `initializeTransitionContext(...)` method. On the other hand, I like that I can read immediately what the setup does.